src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -1004,10 +1002,28 @@ private Video makeVideo(Imp imp, RubiconVideoParams rubiconVideoParams, String r
 
         return video.toBuilder()
                 .ext(mapper.mapper().valueToTree(
-                        RubiconVideoExt.of(skip, skipDelay, RubiconVideoExtRp.of(resolvedSizeId), videoType)))
+                        RubiconVideoExt.of(skip,
+                                skipDelay,
+                                resolvedSizeId != null ? RubiconVideoExtRp.of(resolvedSizeId) : null,
+                                videoType)))
                 .build();
     }
 
+    private Integer resolveSizeId(RubiconVideoParams rubiconVideoParams, Imp imp, String referer) {
+        if (!useVideoSizeLogic) {
+            return null;
+        }
```

Comment:
I'd rather move this logic out of method, and call the method if the `useVideoSizeLogic` is true